test/decode.js

```diff
@@ -34,6 +34,22 @@ describe('decode', function() {
   it('parses url-encoded payload into a buffer', function() {
     var buffer = dataUri.decode('data:,hello%20world');
     expect(buffer.toString('ASCII')).to.equal('hello world');
+    buffer = dataUri.decode('data:;base64,aGVsbG8gd29ybGQ=');
+    expect(buffer.toString('ASCII')).to.equal('hello world');
+    buffer = dataUri.decode('data:;charset=iso-8859,hello%20world');
+    expect(buffer.toString('ASCII')).to.equal('hello world');
+  });
+
+  it('detects the media type', function() {
+    var buffer = dataUri.decode('data:,hello%20world');
+    expect(buffer.mediatype).to.equal('text/plain;charset=US-ASCII');
+    buffer = dataUri.decode('data:text/plain,hello%20world');
+    expect(buffer.mediatype).to.equal('text/plain');
+    buffer = dataUri.decode(
+      'data:text/plain;charset=iso-8859;base64,aGVsbG8gd29ybGQ=');
+    expect(buffer.mediatype).to.equal('text/plain;charset=iso-8859');
+    buffer = dataUri.decode('data:;charset=iso-8859;base64,aGVsbG8gd29ybGQ=');
+    expect(buffer.mediatype).to.equal('text/plain;charset=iso-8859');
```

Comment:
Same comment as above, one assert per test please. Proper unit tests should fail for exactly one reason.